### DISCUSSION TITLE
How to layout containers with embedded childs in a grid/custom  layout?

### DISCUSSION THREAD

**Discussion by MeAriJit** [clientIO/joint#2155] [Q&A]
### Introduction

I have multiple containers and these containers have multiple custom child components embedded in it. 
My purpose is to layout these containers(child elements fits inside) in gridlayout/or any layout. 

Issue: 
  layout.GridLayout.layout(graph,{
    columns:2
  })
This is messing up my  layouting of containers. 

Any solutions to this? Or my approach is not correct.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
You can use the [parentRelative](https://resources.jointjs.com/docs/rappid/v3.6/layout.GridLayout.html) option of the `GridLayout`.
Make sure to call the layout function for each container's children separately.

Note, this does not prevent the children from overlapping the containers. And if you resize the containers to fit the children, it does not prevent containers from overlapping each other. For ultimate solution you'd have to use a combination of layouts as shown for example in [Layouts](https://www.jointjs.com/demos/jointjs-layouts) or [Corporate OrgChart](https://www.jointjs.com/demos/corporate-organizational-chart) demos.

**Reply by MeAriJit** [reply]
Got it. 
But what do you mean by Make sure to call the layout function for each container's children separately." 
Can you give an example?

**Reply by kumilingus** [reply]
`containers.forEach(container => callGridLayout(container.getEmbeddedCells());`

**Reply by MeAriJit** [reply]
I have predefined positions for my child elements(or take an example any elements(with positions) without containers). 
If I use grid layout why it's positions got affected? 
Why can't we only make the container follow gridlayout/otherlayout?

**Reply by kumilingus** [reply]
You can. It was not clear from the question just what you want to achieve.

```js
const containers = graph.getElements().filter(el => !el.isEmbedded());
// 1. the layout function accepts also an array of elements
// 2. use the `deep` option to preserve the relative position of children relative to their parents
joint.layout.GridLayout.layout(containers, { deep: true });
````
